Classes/Command/SupervisorCommandController.php

```diff
@@ -56,7 +49,74 @@ public function createCommand(): void
                 );
             }
         }
+    }
+
+    /**
+     * Starts all programs of configured groups
+     */
+    public function startGroupsCommand(): void
+    {
+        $groups = $this->getGroups();
+        array_walk($groups, [$this, 'runSupervisorCommand'], 'start');
+    }
+
+    /**
+     * Stops all programs of configured groups
+     */
+    public function stopGroupsCommand(): void
+    {
+        $groups = $this->getGroups();
+        array_walk($groups, [$this, 'runSupervisorCommand'], 'stop');
+    }
+
+    /**
+     * Reloads config and starts/stops programs accordingly
+     */
+    public function updateGroupsCommand(): void
+    {
+        $groups = $this->getGroups();
+        array_walk($groups, [$this, 'runSupervisorCommand'], 'update');
+    }
+
+    protected function runSupervisorCommand(Model\Group $group, int $key, string $action): bool
+    {
+        $output = [];
+        $command = sprintf('supervisorctl %s %s%s 2>&1',escapeshellarg($action), escapeshellarg($group->getName()), $action !== 'update' ? ':' : '');
+        $output = exec($command, $output, $result);
```

Comment:
```suggestion
        exec($command, $output, $result);
```
Return value is the last returned line at best, so better not use it at all but rely on the reference argument. Couldn't tell by heart if using the same variable for all arguments (`$output = exec($command, $output)`) would fill the $output with the last line or with every line.

Reply:
Indeed. Output was only one line. It doesn't matter that the return value itself is is the reference arg because it is evaluated last and so the whole array with all lines gets overwritten with the return value.